pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/operands/FilterOperand.java

```diff
@@ -95,6 +114,36 @@ public Boolean apply(Object[] row) {
   @Override
   public abstract Boolean apply(Object[] row);
 
+  private static class BooleanInputRef extends FilterOperand {
+    private final RexExpression.InputRef _inputRef;
+
+    public BooleanInputRef(RexExpression.InputRef inputRef, DataSchema dataSchema) {
+      Preconditions.checkState(dataSchema.getColumnDataType(inputRef.getIndex())
+          == DataSchema.ColumnDataType.BOOLEAN);
+      _inputRef = inputRef;
+    }
+
+    @Override
+    public Boolean apply(Object[] row) {
+      return row[_inputRef.getIndex()] instanceof Integer ? (Integer) row[_inputRef.getIndex()] > 0
```

Comment:
May not be a big deal but is there no way to avoid `instanceof` check ? Could be costly

Reply:
yeah I wasn't able to find a way to avoid this in this PR. but I have another issue open
basically the idea is:
- currently datatable returns the row as stored data type and it doesn't convert it back to the original data type. for example for data that are timestamp and boolean, we convert them in the result table step. 
- this should not be the behavior for data block. as the read out to object[] step reside in datablockutils. it should return the proper columndatatype instead. 

once this issue is addressed we no longer need this conversion (in fact we no longer need conversion of any kind)

Reply:
rebase done